### PR TITLE
ARROW-15690: [Dev] Update GitHub Actions workflows that hardcode master as default

### PR DESCRIPTION
**Pull request by nealrichardson**
Included a few other master-related cleanups I saw

### CODE REVIEW

File: .github/workflows/dev_pr.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ on:
   # TODO: Enable this when eps1lon/actions-label-merge-conflict is available.
   # push:
   #   branches:
-  #     - master
+  #     - [master, main]
```
Comment (nealrichardson) [inline .github/workflows/dev_pr.yml:24]:
```suggestion
  #     - master
  #     - main
```

**Comment by nealrichardson** [comment]
I'll back out the R windows job changes for now, they're unrelated (they're about the r-lib/actions tag, not our master branch). It looks like pacman isn't on the path in the v2 workflows (https://github.com/r-lib/actions/compare/master...v2-branch#diff-3e2a1a7f497c1884dc161d6e07fdc58e4853eb91bf13034122e3814b501df0d5L382), I'll follow up with that upstream.

File: ci/scripts/install_kartothek.sh [hunk 2]
```diff
@@ -28,9 +28,7 @@ karthothek=$1
 target=$2
 
 git clone --recurse-submodules https://github.com/JDASoftwareGroup/kartothek "${target}"
-if [ "${kartothek}" = "master" ]; then
```
Comment (pitrou) [inline ci/scripts/install_kartothek.sh:31]: Did you check that no existing CI job was passing "master" to this script?

File: ci/scripts/install_turbodbc.sh [hunk 3]
```diff
@@ -28,9 +28,7 @@ turbodbc=$1
 target=$2
 
 git clone --recurse-submodules https://github.com/blue-yonder/turbodbc "${target}"
-if [ "${turbodbc}" = "master" ]; then
```
Comment (pitrou) [inline ci/scripts/install_turbodbc.sh:31]: Same question here.

File: ci/scripts/install_kartothek.sh (hunk 2 above)
Comment (pitrou) [reply ci/scripts/install_kartothek.sh:31]: I see, it's simply redundant with the `else` branch below?

Comment (nealrichardson) [reply ci/scripts/install_kartothek.sh:31]: Correct

Comment (kszucs) [reply ci/scripts/install_kartothek.sh:31]: Yes, we can safely remove this branch.

**Comment by kszucs** [comment]
@nealrichardson it looks good to me, but the R builds are failing.

**Review by kszucs — APPROVED** [review]

**Pull request closed without merging**